Source/Core/VideoBackends/Software/Tev.h

```diff
@@ -3,10 +3,90 @@
 
 #pragma once
 
+#include <array>
+
+#include "Common/EnumMap.h"
 #include "VideoCommon/BPMemory.h"
 
 class Tev
 {
+  struct TevColor
+  {
+    constexpr TevColor() = default;
+    constexpr explicit TevColor(s16 a, s16 b, s16 g, s16 r) : a(a), b(b), g(g), r(r) {}
+
+    s16 a = 0;
+    s16 b = 0;
+    s16 g = 0;
+    s16 r = 0;
+
+    constexpr static TevColor All(s16 value) { return TevColor(value, value, value, value); }
+
+    constexpr s16& operator[](int index)
```

Comment:
`constexpr s16& operator[](int index) const` ?

Reply:
That would need to be `constexpr const s16& operator[](int index) const`, but it's also assigned to in some cases so that won't work.  Theoretically both could be added, but there doesn't seem to be a case where it's needed.